pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -74,6 +74,7 @@ type Service struct {
 type context struct {
 	outboundDispatcher dispatcher.Outbound
 	didCreator         did.Creator
+	thid               string
```

Comment:
this does not belong here.

Reply:
we basically need to find calculate threadID/ID out of state machine handle and the pass it as a parameter to state machine for persistence purposes. Context is internal struct. Other place is to add in DIDCommMsg struct, but that doesn't seem ok. 

Any suggestions ?